pygmt/figure.py

```diff
@@ -26,6 +22,32 @@
 # This is needed for the sphinx-gallery scraper in pygmt/sphinx_gallery.py
 SHOWED_FIGURES = []
 
+# Configuration options for Jupyter notebook support
+SHOW_CONFIG = {
+    "external": True,  # Open in an external viewer
+    "notebook": True,  # Notebook display
+    "dpi": 200,  # default DPI
+}
```

Comment:
Note that `xarray` has a `set_options` API (see http://xarray.pydata.org/en/v0.16.2/generated/xarray.set_options.html and code at https://github.com/pydata/xarray/blob/v0.16.2/xarray/core/options.py) that we could mimic.

Reply:
> Is there a way to avoid the use of a global `SHOW_CONFIG` variable? How about setting these variables in the `__init__` of the `pygmt.Figure` class as `self.is_external` and `self.is_notebook` (or is there some reason I'm missing)?

I think moving these settings into `__init__` means that we have to re-detect the environment for each `Figure()` instance.

Reply:
> > Is there a way to avoid the use of a global `SHOW_CONFIG` variable? How about setting these variables in the `__init__` of the `pygmt.Figure` class as `self.is_external` and `self.is_notebook` (or is there some reason I'm missing)?
> 
> I think moving these settings into `__init__` means that we have to re-detect the environment for each `Figure()` instance.

Right, let's not put it in `__init__.py` then. But should we keep this in `figure.py`? Or do we want to consider moving some of this logic into an `options.py` file, i.e. in line with my comment at https://github.com/GenericMappingTools/pygmt/pull/529/files#r569876074 on adapting `xarray`'s `set_options` API?

Reply:
This PR provides a `pygmt.set_display()` function. Now it can select the display method ("notebook" or "external"), but it's easy to allow it to set other configurations (e.g., *width*, *dpi*) for "notebook" display.

Reply:
> This PR provides a `pygmt.set_display()` function. Now it can select the display method ("notebook" or "external")

Ok with this.

> but it's easy to allow it to set other configurations (e.g., _width_, _dpi_) for "notebook" display.

Can we postpone the `dpi` and `width` configuration setting in a separate PR (i.e. keep this PR solely focused on automatic notebook or external display)? More to keep the scope of this PR small, but also because the `dpi` setting seems like a parallel issue outside of resolving #269 (i.e. needs more discussion on what a suitable default `dpi` is).